### PR TITLE
Feat/create report endpoint

### PR DESCRIPTION
**Pull request by fatjan**
The endpoint to create report

Screenshot 2024-01-29 at 10 01 46

Screenshot 2024-01-29 at 11 42 18

Local tests passed:
Screenshot 2024-01-30 at 12 57 12

### CODE REVIEW

**Comment by fatjan** [comment]
I am tired fixing the failed test. Set this PR back to draft, will continue again later. Should have only created a PR once all tests passed.

File: app/main/model/report.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    __tablename__ = "report"
+
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    user_id = db.Column(db.Integer, db.ForeignKey("user.id"), nullable=False)
+    public_id = db.Column(db.String(100), unique=True, nullable=False)
+    type = db.Column(db.String(100), nullable=False)
+    item_id = db.Column(db.Integer, db.ForeignKey('review.id'), nullable=False)
+    reason = db.Column(db.String(255), nullable=False)
+    created_at = db.Column(db.DateTime, nullable=False)
+    updated_at = db.Column(db.DateTime, nullable=False)
```
Comment (fahmirzldi) [inline app/main/model/report.py:20]: Should we add some "status" column to help check whether report has been processed or not?

Comment (fahmirzldi) [reply app/main/model/report.py:20]: I think

- Received
- Accepted
- Rejected

would be enough as "prccessed" would be unclear

Comment (fatjan) [reply app/main/model/report.py:20]: noted

Comment (fatjan) [reply app/main/model/report.py:21]: Updated with the status in the table, kindly review it again, thank you

File: app/main/model/report.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    def save(self):
+        db.session.add(self)
+        db.session.commit()
+
+    def create_report(self, data):
+        try:
+            self.public_id = str(uuid.uuid4())
+            self.user_id = data.get("user_id")
+            self.type = data.get("type")
+            self.status = data.get("status", "received")
```
[the remark this answers is not in the document]
Comment (fatjan) [reply app/main/model/report.py:60]: good question, a newly created report should have `received` as status only. I'll update the PR

[the remark this answers is not in the document]
Comment (fatjan) [reply app/main/model/report.py:60]: yes, there is, we can use set (a tuple)

[the remark this answers is not in the document]
Comment (fatjan) [reply app/main/model/report.py]: we can do what @AchmadQiz did before creating a class with the list of values in it, but it failed when I migrate the db from MySQL to postgresql
I think we can try again to make it work

[the remark this answers is not in the document]
Comment (fahmirzldi) [reply app/main/model/report.py]: Is it the one that use `enum()`?

[the remark this answers is not in the document]
Comment (fatjan) [reply app/main/model/report.py]: yes

**Review by fahmirzldi — APPROVED** [review]

**Pull request merged**